### PR TITLE
Feature/Services2.0

### PR DESCRIPTION
**Pull request by Simonmicro**
- Refactored @SeanReg service system
- Added example service
- Added memory monitor service
- Added GPS service (untested!)
- Cleanup

### CODE REVIEW

File: src/services/OswServiceTasks.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#ifdef GPS_EDITION
+OswServiceTaskGPS gps;
+#endif
+//OswServiceTaskExample example;
+#ifdef DEBUG
+OswServiceTaskMemMonitor memory;
+#endif
+}  // namespace OswServiceAllTasks
+
+const unsigned char oswServiceTasksCount = 0 + SERVICE_BLE_COMPANION + SERVICE_GPS + SERVICE_MEMMONITOR;
```
Comment (uvwxy) [inline src/services/OswServiceTasks.cpp:33]: I was thinking about how we would do this one 👍 
too simple :)

File: src/services/OswServiceTaskMemMonitor.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#include "./services/OswServiceTaskMemMonitor.h"
+
+#include "osw_hal.h"
+#include "services/OswServiceManager.h"
+
+void OswServiceTaskMemMonitor::setup(OswHal* hal) { OswServiceTask::setup(hal); }
+
+void OswServiceTaskMemMonitor::loop(OswHal* hal) {
+  unsigned core0 = uxTaskGetStackHighWaterMark(nullptr);
```
Comment (uvwxy) [inline src/services/OswServiceTaskMemMonitor.cpp:9]: I think I could benefit from some comments to understand this

**Review by uvwxy — APPROVED** [review]

File: src/services/OswServiceTasks.cpp (hunk 1 above)
Comment (Simonmicro) [reply src/services/OswServiceTasks.cpp:33]: :grin: - is this comment done?

File: src/services/OswServiceTaskMemMonitor.cpp (hunk 2 above)
Comment (Simonmicro) [reply src/services/OswServiceTaskMemMonitor.cpp:9]: OK - comes soon :tm:

**Pull request merged**